src/commands/CommandMusicPlay.java

```diff
@@ -35,7 +48,30 @@ public void action(){
 			}
 
 			if(getContent() != null){
-				MusicManager.get().loadTrack(this, getContent());
+				String id;
+				String source;
+				if (!isUrl(getContent())){
+					YouTube youtube = new YouTube.Builder(new NetHttpTransport(), new JacksonFactory(), new HttpRequestInitializer() {
+						public void initialize(HttpRequest request) throws IOException {
+						}
+					}).setApplicationName("Discord Bot").build();
+					try {
+						YouTube.Search.List search = youtube.search().list("snippet");
+						search.setMaxResults((long) 1);
+						search.setQ(getContent());
+						search.setKey(Environment.getVar("YOUTUBE_TOKEN"));
+						SearchListResponse response = search.execute();
+						id = response.getItems().get(0).getId().getVideoId();
+						System.out.print(id);
+						source = "https://www.youtube.com/watch?v=" + id;
+						MusicManager.get().loadTrack(this, source);
+					} catch (IOException e) {
+						e.printStackTrace();
+					}
+				}else{
+					source = getContent();
+					MusicManager.get().loadTrack(this, source);
```

Comment:
I see that `loadTrack` is called here and at line 67 using the exact same parameters - might as well remove the one at line 67 and move this one after the end of the else block, this way we'd remove a bit of code duplication.

This change would probably mean that your `try catch(IOException)` (at lines 58-70) would enclose the whole content (lines 51-74). This way, if an error happens while searching using plain text, it won't also crash the loading of the song (because it shouldn't try to load a track with a broken source).